nautobot/extras/plugins/__init__.py

```diff
@@ -322,19 +321,60 @@ def __init__(self, link, title, icon_class, color=None, permissions=None):
             self.color = color
 
 
-def register_menu_items(section_name, class_list):
+def register_plugin_menu_items(section_name, menu_items):
     """
     Register a list of PluginMenuItem instances for a given menu section (e.g. plugin name)
     """
-    # Validation
-    for menu_link in class_list:
-        if not isinstance(menu_link, PluginMenuItem):
-            raise TypeError(f"{menu_link} must be an instance of extras.plugins.PluginMenuItem")
-        for button in menu_link.buttons:
-            if not isinstance(button, PluginMenuButton):
-                raise TypeError(f"{button} must be an instance of extras.plugins.PluginMenuButton")
-
-    registry["plugin_menu_items"][section_name] = class_list
+    new_menu_items = []
+    new_menu_item_weight = 100
+    for menu_item in menu_items:
+        if isinstance(menu_item, PluginMenuItem):
+            # translate old-style plugin menu definitions into the new nav-menu items and buttons
+
+            new_menu_button_weight = 100
+            new_menu_buttons = []
+            for button in menu_item.buttons:
+                new_menu_buttons.append(
+                    NavMenuButton(
+                        link=button.link,
+                        title=button.title,
+                        icon_class=button.icon_class,
+                        button_class=button.color,
```

Comment:
Does this work? I'd have thought we'd need to have some sort of translation/mapping from `color` to `button_class`?

Reply:
The color needs to be defined by using a bootstrap button type like `btn-success` for both plugins and core app so i dont think there is any translation needed.

Reply:
Ah, you're right. I was misremembering that the button color choices were actual HTML color values, but you're right that they're also bootstrap classes.